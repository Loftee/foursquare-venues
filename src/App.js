import React, {Component} from 'react';
import {fetchVenues} from './services/services';
import VenueList from './components/VenueList';
import Search from './components/Search';
import VenueDetails from './components/VenueDetails';
import * as constants from './constants';

import './App.scss';

class App extends Component {
    constructor() {
        super();
        this.state = {
            venues: [],
            searchText: '',
            selectedVenue: {}
        }
    }

    searchChange = event => {
        const searchText = event.target.value;
        this.setState(prevState => ({
            ...prevState,
            searchText: searchText
        }));
    };

    search = () => {
        const searchQuery = constants.VENUE_API + '&query=' + this.state.searchText.toLowerCase().replace(' ', '+');
        fetchVenues(searchQuery)
            .then(res => {
                const venueList = res ? res : [];
                this.setState(prevState => ({
                    ...prevState,
                    venues: venueList
                }));
            });
    };

    selectVenue = id => {
        const venue = this.state.venues.filter(venue => venue.id === id);
        this.setState(prevState => ({
            ...prevState,
            selectedVenue: venue[0]
        }));
    };

    render() {
        return (
            <div className="app">
                <h1>Search for locations near Oxford Circus</h1>
                <Search searchText={this.state.searchText} searchChange={this.searchChange} search={this.search}/>
                {this.state.venues.length > 0 &&
                <VenueList {...this.state} selectVenue={this.selectVenue}/>
                }
                {this.state.selectedVenue.name &&
                <VenueDetails selectedVenue={this.state.selectedVenue}/>
                }
            </div>
        );
    }
}

export default App;
