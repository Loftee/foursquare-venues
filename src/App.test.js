import React from 'react';
import {shallow} from 'enzyme';
import App from './App';
import {testVenue} from "./testData";

describe('App', () => {
    const app = shallow(<App/>);
    app.setState({
        venues: [testVenue],
        searchText: '',
        selectedVenue: testVenue
    });

    it('renders correctly', () => {
        expect(app).toMatchSnapshot();
    });

    it('renders VenueList component', () => {
        expect(app.find('VenueList').exists()).toBe(true);
    });

    it('renders Search component', () => {
        expect(app.find('Search').exists()).toBe(true);
    });

    it('renders VenueDetails component', () => {
        expect(app.find('VenueDetails').exists()).toBe(true);
    });
});
