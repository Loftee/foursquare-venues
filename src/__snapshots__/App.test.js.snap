// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
<div
  className="app"
>
  <h1>
    Search for locations near Oxford Circus
  </h1>
  <Search
    search={[Function]}
    searchChange={[Function]}
    searchText=""
  />
  <VenueList
    searchText=""
    selectVenue={[Function]}
    selectedVenue={
      Object {
        "id": "4c3d6ed94bc9a5939b07d471",
        "location": Object {
          "address": "4 Great Portland St",
          "cc": "GB",
          "city": "London",
          "country": "United Kingdom",
          "distance": 42,
          "formattedAddress": Array [
            "4 Great Portland St",
            "London",
            "Greater London",
            "W1W 8QJ",
            "United Kingdom",
          ],
          "labeledLatLngs": Array [
            Object {
              "label": "display",
              "lat": 51.51573111967288,
              "lng": -0.14068655684295295,
            },
          ],
          "lat": 51.51573111967288,
          "lng": -0.14068655684295295,
          "postalCode": "W1W 8QJ",
          "state": "Greater London",
        },
        "name": "Costa Coffee",
      }
    }
    venues={
      Array [
        Object {
          "id": "4c3d6ed94bc9a5939b07d471",
          "location": Object {
            "address": "4 Great Portland St",
            "cc": "GB",
            "city": "London",
            "country": "United Kingdom",
            "distance": 42,
            "formattedAddress": Array [
              "4 Great Portland St",
              "London",
              "Greater London",
              "W1W 8QJ",
              "United Kingdom",
            ],
            "labeledLatLngs": Array [
              Object {
                "label": "display",
                "lat": 51.51573111967288,
                "lng": -0.14068655684295295,
              },
            ],
            "lat": 51.51573111967288,
            "lng": -0.14068655684295295,
            "postalCode": "W1W 8QJ",
            "state": "Greater London",
          },
          "name": "Costa Coffee",
        },
      ]
    }
  />
  <VenueDetails
    selectedVenue={
      Object {
        "id": "4c3d6ed94bc9a5939b07d471",
        "location": Object {
          "address": "4 Great Portland St",
          "cc": "GB",
          "city": "London",
          "country": "United Kingdom",
          "distance": 42,
          "formattedAddress": Array [
            "4 Great Portland St",
            "London",
            "Greater London",
            "W1W 8QJ",
            "United Kingdom",
          ],
          "labeledLatLngs": Array [
            Object {
              "label": "display",
              "lat": 51.51573111967288,
              "lng": -0.14068655684295295,
            },
          ],
          "lat": 51.51573111967288,
          "lng": -0.14068655684295295,
          "postalCode": "W1W 8QJ",
          "state": "Greater London",
        },
        "name": "Costa Coffee",
      }
    }
  />
</div>
`;
